tests/Adapters/Microsoft.Bot.Builder.Adapters.Facebook.Tests/FacebookAdapterTests.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Schema;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Bot.Builder.Adapters.Facebook.Tests
+{
+    public class FacebookAdapterTests
+    {
+        private readonly FacebookAdapterOptions _testOptions = new FacebookAdapterOptions("Test", "Test", "Test");
+
+        [Fact]
+        public void ConstructorShouldFailWithNullClient()
+        {
+            Assert.Throws<ArgumentNullException>(() => { new FacebookAdapter((FacebookClientWrapper)null); });
+        }
+
+        [Fact]
+        public void ConstructorWithArgumentsShouldSucceed()
+        {
+            Assert.NotNull(new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object));
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldFailWithNullConversationReference()
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+
+            Task BotsLogic(ITurnContext turnContext, CancellationToken cancellationToken)
+            {
+                return Task.CompletedTask;
+            }
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(null, BotsLogic); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldFailWithNullLogic()
+        {
+            var facebookAdapter = new FacebookAdapter(new Mock<FacebookClientWrapper>(_testOptions).Object);
+            var conversationReference = new ConversationReference();
+
+            await Assert.ThrowsAsync<ArgumentNullException>(async () => { await facebookAdapter.ContinueConversationAsync(conversationReference, null); });
+        }
+
+        [Fact]
+        public async void ContinueConversationAsyncShouldSucceed()
```

Comment:
I _think_ positive tests tend to go before negative ones. But, not sure that's a rule.